packages/asana/src/Adaptor.js

```diff
@@ -233,39 +168,122 @@ export function createTask(params, callback) {
  */
 export function upsertTask(projectGid, params, callback) {
   return state => {
-    const resolvedProjectGid = expandReferences(projectGid)(state);
-    const { externalId, data } = expandReferences(params)(state);
-
-    const { apiVersion, token } = state.configuration;
-
-    const url = `https://app.asana.com/api/${apiVersion}/projects/${resolvedProjectGid}/tasks`;
-
-    const config = {
-      url,
-      headers: { Authorization: `Bearer ${token}` },
-      params: {
-        opt_fields: `${externalId}`,
-      },
-    };
-
-    return http
-      .get(config)(state)
-      .then(response => {
-        const matchingTask = response.data.data.find(
+    const [resolvedProjectGid, resolvedParams] = expandReferences(
+      state,
+      projectGid,
+      params
+    );
+    const { externalId, data } = resolvedParams;
```

Comment:
Little tip - you should be able to deconstruct these in-line in your expand statement:
```
 const [resolvedProjectGid, { externalId, data }] = expandReferences(
      state,
      projectGid,
      params
    );
```

Reply:
Thanks, this is good 👍🏽